Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -796,11 +813,107 @@ namespace xr
+
+            // Next check for updated planes, and update their pose and polygon or create a new plane if it does not yet exist.
+            ArFrame_getUpdatedTrackables(session, frame, AR_TRACKABLE_PLANE, trackableList);            
+            int32_t size{};
+            ArTrackableList_getSize(session, trackableList, &size);
+            for (int i = 0; i < size; i++)
+            {
+                // Get the plane.
+                ArTrackable* trackable;
+                ArTrackableList_acquireItem(session, trackableList, i, &trackable);
+                auto planeTrackable = reinterpret_cast<ArPlane*>(trackable);
+
+                // Check if this plane has been subsumed. If so skip it as we are about to delete this plane.
+                ArPlane* subsumingPlane = nullptr;
+                ArPlane_acquireSubsumedBy(session, planeTrackable, &subsumingPlane);
+                if (subsumingPlane != nullptr)
+                {
+                    ArTrackable_release(trackable);
+                    ArTrackable_release(reinterpret_cast<ArTrackable*>(subsumingPlane));
+                    continue;
+                }
+
+                // Get the center pose.
+                float rawPose[7]{};
+                ArPlane_getCenterPose(session, planeTrackable, tempPose);
+                ArPose_getPoseRaw(session, tempPose, rawPose);
+
+                // Dynamically allocate the polygon array, and fill it in.
+                int32_t polygonSize;
+                ArPlane_getPolygonSize(session, planeTrackable, &polygonSize);
+                float* polygon = reinterpret_cast<float*>(malloc(sizeof(float) * polygonSize));
+
+                // Get the polygon.
+                ArPlane_getPolygon(session, planeTrackable, polygon);
+
+                // Update the existing plane if it exists, otherwise create a new plane, and add it to our list of planes.
+                auto planeIterator = existingPlanes.find(reinterpret_cast<NativePlaneIdentifier>(trackable));
+                if (planeIterator != existingPlanes.end())
+                {
+                    UpdateExistingPlane(planeIterator->second, rawPose, polygon, polygonSize);
+                }
+                else
+                {
+                    // This is a new plane, create it and initialize its values.
+                    Plane plane{};
+                    plane.NativePlaneId = reinterpret_cast<NativePlaneIdentifier>(trackable);
+                    UpdateExistingPlane(&plane, rawPose, polygon, polygonSize);
+                    newPlanes.push_back(plane);
+                }
+            }
+        }
+
+        bool CheckIfPlaneWasUpdated(Plane* existingPlane, float* newPolygon, size_t newPolygonSize, Pose& newCenter)
+        {
+            // First check if the center has changed, or the polygon size has changed.
+            if (existingPlane->Center != newCenter
+                || existingPlane->PolygonSize * 2 != newPolygonSize)
```

Comment:
What is the `* 2`?

Reply:
I got rid of this.  This is because the existingPlane->Polygonsize is the number of verts and newPolygonSize was the size of the array.  I agree it's confusing so I removed this and just check for the vector size now.